File: PYTHON/Program/Garage.py
# Saya Rifa Sania NIM 2206697 mengerjakan Latihan Praktikum 4 dalam mata kuliah Desain Pemrograman Berorientasi Objek
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin...

class Garage:
    __namaGarasi = ""
    __luasGarasi = ""
    __daftarKendaraan = []
    __parkingLot = None

    def __init__(self):
        self.__namaGarasi = ""
        self.__luasGarasi = ""
        self.__daftarKendaraan = []

    def __init__(self, namaGarasi = "", luasGarasi = "", daftarKendaraan = [], parkingLot = None):
        self.__namaGarasi = namaGarasi
        self.__luasGarasi = luasGarasi
        self.__daftarKendaraan = daftarKendaraan
        self.__parkingLot = parkingLot

    def setNamaGarasi(self, namaGarasi):
        self.__namaGarasi = namaGarasi

    def getNamaGarasi(self):
        return self.__namaGarasi

    def setLuasGarasi(self, luasGarasi):
        self.__luasGarasi = luasGarasi

    def getLuasGarasi(self):
        return self.__luasGarasi

    def setDaftarKendaraan(self, daftarKendaraan):
        self.__daftarKendaraan = daftarKendaraan

    def getDaftarKendaraan(self):
        return self.__daftarKendaraan 

    def setParkingLot(self, parkingLot):
        self.__parkingLot = parkingLot

    def getParkingLot(self):
        return self.__parkingLot     

    def tambahKendaraan(self, kendaraan):
        self.__daftarKendaraan.append(kendaraan)    

File: CPP/Program/Garage.cpp
#pragma once
#include <iostream>
#include <string>
#include <vector>
#include "Vehicle.cpp"
#include "ParkingLot.cpp"

using namespace std;

class Garage{
    private:
        string namaGarasi;
        string luasGarasi;
        vector<Vehicle*> daftarKendaraan;
        ParkingLot parkingLot;

    public:
        Garage(){
        }

        Garage(string namaGarasi){
            this->namaGarasi = namaGarasi;
            luasGarasi = "";
        }

        Garage(string plaNomor, string luasGarasi){
            this->namaGarasi = plaNomor;
            this->luasGarasi = luasGarasi;
        }

        Garage(string namaGarasi, string luasGarasi, vector<Vehicle*> daftarKendaraan){
            this->namaGarasi = namaGarasi;
            this->luasGarasi = luasGarasi;
            this->daftarKendaraan = daftarKendaraan;
        }

        void setNamaGarasi(string namaGarasi){
            this->namaGarasi = namaGarasi;
        }

        string getNamaGarasi(){
            return namaGarasi;
        }

        void setLuasGarasi(string luasGarasi){
            this->luasGarasi = luasGarasi;
        }

        string getLuasGarasi(){
            return luasGarasi;
        }
        
        void setDaftarKendaraan(vector<Vehicle*> daftarKendaraan){
            this->daftarKendaraan = daftarKendaraan;
        }

        vector<Vehicle*> getDaftarKendaraan(){
            return daftarKendaraan;
        }

        void setParkingLot(ParkingLot parkingLot) {
            this->parkingLot = parkingLot;
        }

        ParkingLot getParkingLot() {
            return parkingLot;
        }

        void tambahKendaraan(Vehicle* kendaraan) {
            daftarKendaraan.push_back(kendaraan);
        }

        ~Garage(){
        }
};